### PR TITLE
Replace IODbstorage to LevelDBStorage

### PR DESCRIPTION
**Pull request by oaleksandrov-ip**

### CODE REVIEW

File: sdk/src/main/scala/com/horizen/storage/leveldb/VersionedLevelDbStorageAdapter.scala [hunk 1]
```diff
@@ -48,25 +54,41 @@ class VersionedLevelDbStorageAdapter(pathToDB: String, keepVersions: Int) extend
     //key for storing version shall not be used as key in any key-value pair in VersionedLDBKVStore
     require(!toUpdateAsScala.exists(pair => pair.getKey == version) && !toRemoveAsScala.contains(version))
 
+    //@TODO added for compatibility with LSMStore, probably interface shall be changed to work with map collection for toUpdate/toremove
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/storage/leveldb/VersionedLevelDbStorageAdapter.scala:57]: agree

File: sdk/src/test/java/com/horizen/storage/StorageTest.java [hunk 2]
```diff
@@ -1,72 +1,159 @@
...
     public void testStorage() {
-        Storage s = new IODBStoreAdapter(storageFixture.getStore());
+        VersionedLevelDbStorageAdapter s = storageFixture.getStorage();
         assertTrue("Storage expected to be empty.", s.isEmpty());
 
-        ByteArrayWrapper version1 = storageFixture.getVersion();
-        List<Pair<ByteArrayWrapper,ByteArrayWrapper>> u1 = storageFixture.getKeyValueList(3);
+        assertFalse("Version of empty storage shall be none", s.lastVersionID().isPresent());
+
+        ByteArrayWrapper version0 = storageFixture.getVersion();
```
Comment (i-Alex) [inline sdk/src/test/java/com/horizen/storage/StorageTest.java:26]: I would prefer to add some comments before each logical section like:
// Test 1: add entries to empty storage
......
//  Test 2: add entries to non-empty storage
.....
// Test 3: update existing entries
and so on. Just to follow the test cases faster.

File: sdk/src/test/java/com/horizen/storage/StorageTest.java [hunk 3]
```diff
@@ -1,72 +1,159 @@
...
+
 
         ByteArrayWrapper version3 = storageFixture.getVersion();
         List<ByteArrayWrapper> r = new ArrayList<>();
 
         for (Pair<ByteArrayWrapper,ByteArrayWrapper> i : u2.subList(1,3))
             r.add(i.getKey());
 
-        s.update(version3, new ArrayList<Pair<ByteArrayWrapper,ByteArrayWrapper>>(), r);
+        s.update(version3, new ArrayList<>(), r);
```
Comment (i-Alex) [inline sdk/src/test/java/com/horizen/storage/StorageTest.java:68]: Let's add the test case where we try to remove non-existing elements - should be successful.

File: sdk/src/test/java/com/horizen/storage/StorageTest.java [hunk 4]
```diff
@@ -1,72 +1,159 @@
...
+
+        assertFalse("Storage expected to be not empty.", s.isEmpty());
+        assertEquals("Storage must contain 3 items.", firstUpdateSize, s.getAll().size());
+        assertEquals("Storage must contain value for key - " + u1.get(0).getKey(), u1.get(0).getValue(), s.get(u1.get(0).getKey()).get());
+        assertTrue("Storage must contain same elements as sample.", u1.containsAll(s.getAll()));
+        assertEquals("Storage must have specified last version.", version1, s.lastVersionID().get());
+        assertEquals("Storage must have two versions", 2, s.rollbackVersions().size());
+        assertTrue("Storage must have specified versions", s.rollbackVersions().containsAll(Arrays.asList(version0, version1)));
+
+        s.close();
```
Comment (i-Alex) [inline sdk/src/test/java/com/horizen/storage/StorageTest.java:148]: Probably, we can put re-open test case at the end of the method above, to not to duplicate the code of data population. Not significant.

File: sdk/src/test/scala/com/horizen/fixtures/StoreFixture.scala [hunk 5]
```diff
@@ -13,32 +14,31 @@ import com.horizen.utils.ByteArrayWrapper
...
   Runtime.getRuntime.addShutdownHook(new Thread() {
     override def run(): Unit = {
-      for (s <- stores) {
-        s._2.close()
-        for (f <- s._1.listFiles())
-          f.delete()
-        s._1.delete()
-      }
+      storages.foreach(_.close())
+      storages.foreach(_.close())
```
Comment (i-Alex) [inline sdk/src/test/scala/com/horizen/fixtures/StoreFixture.scala:28]: duplicate?

File: sdk/src/test/java/com/horizen/storage/StorageTest.java (hunk 2 above)
Comment (oaleksandrov-ip) [reply sdk/src/test/java/com/horizen/storage/StorageTest.java:26]: done

File: sdk/src/test/java/com/horizen/storage/StorageTest.java (hunk 3 above)
Comment (oaleksandrov-ip) [reply sdk/src/test/java/com/horizen/storage/StorageTest.java:68]: done

File: sdk/src/test/java/com/horizen/storage/StorageTest.java (hunk 4 above)
Comment (oaleksandrov-ip) [reply sdk/src/test/java/com/horizen/storage/StorageTest.java:148]: tests shall be independent from each other...

File: sdk/src/test/scala/com/horizen/fixtures/StoreFixture.scala (hunk 5 above)
Comment (oaleksandrov-ip) [reply sdk/src/test/scala/com/horizen/fixtures/StoreFixture.scala:28]: fixed

**Pull request merged**